examples/vm_scheduling/rule_based_algorithm/first_fit.py

```diff
@@ -0,0 +1,21 @@
+from maro.simulator import Env
+from maro.simulator.scenarios.vm_scheduling.common import Action
+from maro.simulator.scenarios.vm_scheduling import AllocateAction, DecisionPayload, PostponeAction
+
+from rule_based_algorithm import RuleBasedAlgorithm
+
+
+class FirstFit(RuleBasedAlgorithm):
+    def __init__(self):
+        super().__init__()
+
+    def allocate_vm(self, decision_event: DecisionPayload, env: Env) -> AllocateAction:
+        # Use a valid PM based on its order.
+        chosen_idx: int = decision_event.valid_pms[0]
```

Comment:
sort and make sure the first one is the one with the smallest id

Reply:
The given decision_event.valid_pms has been sorted, so I don't sort it again.